### PR TITLE
refactor(itt): move to go based agent

### PR DESCRIPTION
**Pull request by fieldju**
In addition to the unit tests I also did the following UAT.

1. I launched an arm based ec2 instance and installed k3s, helm and then installed the agent and confirmed that the helm chart will auto-detect linux amd64 or arm64.
image
image
image0

2. I installed [Release Candidate 2](https://github.com/armory-io/remote-network-agent-helm-chart/releases/tag/v2.0.0-RC2) based off of this branch into my dev cluster via

```bash
helm upgrade --install armory-rna remote-network-agent \
    --version v2.0.0-RC2 \
    --repo 'https://armory.jfrog.io/artifactory/charts' \
    --set clientId='encrypted:k8s!n:rna-client-credentials!k:client-id' \
    --set clientSecret='encrypted:k8s!n:rna-client-credentials!k:client-secret' \
    --set agentIdentifier=go-uat \
    --namespace armory-rna
```

which uses the 3.+ version of the Agent (go based)

image

image

Then I used the UI to deploy the sample app to it

image

and confirmed that things are working

image

### CODE REVIEW

**Comment by fieldju** [comment]
Something is wrong with the build, I'll check it out.

but the tests do run and pass locally
image

**Comment by fieldju** [comment]
image
image

**Review by tunderwood — APPROVED** [review]

File: templates/deployment.yaml [hunk 1]
```diff
@@ -25,42 +29,100 @@ spec:
...
+              value: "{{ .Values.log.level }}"
+            - name: DISABLE_COLORS
+              value: "{{ .Values.log.disableColor }}"
+            {{- if not (empty .Values.sandboxClusterId) }}
+            - name: agent_sandboxClusterId
+              value: {{ .Values.sandboxClusterId }}
+            {{- end }}
+            - name: oidc_clientId
+              value: "{{ include "armory-remote-network-agent.client-id" . }}"
+            - name: oidc_clientSecret
```
Comment (Greg-At-Armory) [inline templates/deployment.yaml:73]: out of curiosity - shouldnt client id and client secret be loaded from secrets? i mean - simple kubectl get pod would reveal all the environment values, but if that was configured with envFromSecret, it wouldn't be so easily visible. on the other hand - if one can run kubectl get po, there is nothing preventing him from doing kubectl exec and inspecting the actual env ...

**Review by Greg-At-Armory — APPROVED** [review]

File: templates/deployment.yaml (hunk 1 above)
Comment (fieldju) [reply templates/deployment.yaml:73]: Yes you are correct, in the readme and values file we highly encourage the use of encrypted tokens

image

https://github.com/armory-io/remote-network-agent-helm-chart/blob/e9cf9962d86f578d20e69e9ff1e607ea564cdb56/values.yaml#L1-L28

So the value that will be injected here will be 

```
encrypted:secrets-manager!r:us-west-2!s:some-aws-sm-secret-name!k:client-secret
```

or

```
encrypted:secrets-manager!r:us-west-2!s:some-aws-sm-secret-name!k:client-secret
```

:point_up: these are converted to the actual values at run time

**Pull request merged**